src/target/riscv/riscv.c

```diff
@@ -3104,29 +3112,42 @@ static int check_virt_memory_access(struct target *target, target_addr_t address
 static int riscv_read_phys_memory(struct target *target, target_addr_t phys_address,
 			uint32_t size, uint32_t count, uint8_t *buffer)
 {
+	riscv_mem_access_info_t read_info = {
+		.target_address = phys_address,
+		.buffer.read = buffer,
+		.element_size = size,
+		.increment = size,
+		.count = count,
+		.is_read = true,
+		.is_virtual = false,
+	};
 	RISCV_INFO(r);
-	return r->read_memory(target, phys_address, size, count, buffer, size);
+	return r->read_memory(target, read_info);
 }
 
 static int riscv_write_phys_memory(struct target *target, target_addr_t phys_address,
 			uint32_t size, uint32_t count, const uint8_t *buffer)
 {
-	struct target_type *tt = get_target_type(target);
-	if (!tt)
-		return ERROR_FAIL;
-	return tt->write_memory(target, phys_address, size, count, buffer);
+	riscv_mem_access_info_t write_info = {
+		.target_address = phys_address,
+		.buffer.write = buffer,
+		.element_size = size,
+		.increment = size,
+		.count = count,
+		.is_read = false,
+		.is_virtual = false,
+	};
+
+	RISCV_INFO(r);
+	return r->write_memory(target, write_info);
 }
 
-static int riscv_rw_memory(struct target *target, target_addr_t address, uint32_t size,
-		uint32_t count, uint8_t *read_buffer, const uint8_t *write_buffer)
+static int riscv_rw_memory(struct target *target, riscv_mem_access_info_t access)
```

Comment:
Could the `riscv_mem_access_info_t` arguments be `const` (everywhere)?

```suggestion
static int riscv_rw_memory(struct target *target, const riscv_mem_access_info_t access)
```

Reply:
Addressed